django/db/models/sql/query.py

```diff
@@ -104,6 +106,8 @@ def get_child_with_renamed_prefix(prefix, replacement, child):
         child = child.copy()
         if child.name.startswith(prefix + LOOKUP_SEP):
             child.name = child.name.replace(prefix, replacement, 1)
+    elif isinstance(child, QuerySet):
+        pass
```

Comment:
Is it enough to skip querysets? I think this will make the original issue (ticket-34362) unfixed for querysets :thinking: For example:
```python
        qs = Author.objects.annotate(
            poem_book=FilteredRelation(
                "book",
                condition=Q(
                    book__in=Book.objects.filter(book__title="Poem by Alice")
                ),
            ),      
        ).filter(poem_book__isnull=False)
```

Reply:
Indeed, was not as easy as I thought. 
Thanks for the regression test, I will add it to the PR and work on trying to find the proper fix.

Reply:
The lookup inside the rhs's QuerySet was incorrect (`book__title` instead of `title`).
It's working fine now. 
(I added your test case in the PR)

Reply:
I'm surprised that subquery in join condition are even supported on all engines.

I guess the question now becomes what should we do if the queryset need to refer to the filtered relation itself or any one of its right-hand-side lookup? I think that's what @felixxm was referring to.

`OuterRef` are resolved relative to the main model of the query (`Author` in this case) so `OuterRef("book__*")` is allowed and without repointing `book` to `poem_book` like it's the case for pretty much everything else in the condition it will spawn a different join.

```python
qs = Author.objects.annotate(
    poem_book=FilteredRelation(
        "book",
        condition=Q(
            book__in=Book.objects.filter(title=OuterRef("book__title"))
        ),
    ),      
).filter(poem_book__isnull=False)
```

Which should produce the equivalent of

```sql
SELECT *
FROM author
LEFT JOIN book poem_book ON (
    poem_book.author_id = author.id
    AND poem_book.id IN (
        SELECT id
        FROM book
        WHERE title = poem_book.title
    )
)
WHERE poem_book.id IS NOT NULL
```

Maybe it's okay to support queryset but not querysets that have outer references?

Reply:
> I think that's what @felixxm was referring to.

Yes, sorry, I wrote it in a hurry.